File: gallery.c
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 */

#include <stdio.h>
#include <assert.h>

// constraints
#define MAXN 100000

// input data
int N, i;
int X[MAXN];
int Y[MAXN];

int main() {
//  uncomment the following lines if you want to read/write from files
//  freopen("input.txt", "r", stdin);
//  freopen("output.txt", "w", stdout);

    assert(1 == scanf("%d", &N));
    for(i=0; i<N; i++)
        assert(2 == scanf("%d %d", &X[i], &Y[i]));

    // insert your code here
    
    printf("YES\n"); // or "NO\n"
    return 0;
}

File: gallery.pas
{ constraints }
const
    MAXN = 100000;

{ input data }
var
    N, i  : longint;
    X, Y  : array[0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N);
    for i:=0 to N-1 do
        readln(X[i], Y[i]);

    { insert your code here }

    writeln('NO'); { or 'YES' }
end.
